oaebu_workflows/workflows/thoth_telescope.py

```diff
@@ -0,0 +1,331 @@
+        :param dataset_id: the BigQuery dataset id.
+        :param schema_folder: the SQL schema path.
+        :param source_format: the format of the data to load into BigQuery.
+        :param catchup: whether to catchup the DAG or not.
+        :param workflow_id: api workflow id.
+        :param format_specification: The format to use when downloadin Thoth's metadata
+        :param download_file_name: The name of the file to write the ONIX data to
+        :param transform_file_name: The name of the file to write the transformed ONIX data to
+        """
+
+        if airflow_vars is None:
+            airflow_vars = [
+                AirflowVars.DATA_PATH,
+                AirflowVars.PROJECT_ID,
+                AirflowVars.DATA_LOCATION,
+                AirflowVars.DOWNLOAD_BUCKET,
+                AirflowVars.TRANSFORM_BUCKET,
+            ]
+
+        if airflow_conns is None:
+            airflow_conns = []
+
+        self.dataset_description = f"{organisation_name} ONIX feed from Thoth"
+        self.organisation_name = organisation_name
+        self.project_id = project_id
+        self.dataset_id = dataset_id
+        self.download_bucket = download_bucket
+        self.transform_bucket = transform_bucket
+        self.data_location = data_location
+        self.source_format = source_format
+        self.schema_folder = schema_folder
+        self.format_specification = format_specification
+        self.download_file_name = download_file_name
+        self.transform_file_name = transform_file_name
+
+        self.publisher_id = PUBLISHER_IDS.get(organisation_name)
+        if not self.publisher_id:
+            raise AirflowException(f"Publisher ID for organisation '{organisation_name}' not found")
+
+        self.download_folder = None
+        self.transform_folder = None
+
+        super().__init__(
+            dag_id=dag_id,
+            start_date=start_date,
+            schedule_interval=schedule_interval,
+            catchup=catchup,
+            airflow_vars=airflow_vars,
+            airflow_conns=airflow_conns,
+            workflow_id=workflow_id,
+        )
+
+        # self.organisation = organisation
+        self.add_setup_task(self.check_dependencies)
+        self.add_task(self.download)
+        self.add_task(self.upload_downloaded)
+        self.add_task(self.transform)
+        self.add_task(self.upload_transformed)
+        self.add_task(self.bq_load)
+        self.add_task(self.cleanup)
+        self.add_task(self.add_new_dataset_releases)
+
+    def make_release(self, **kwargs) -> ThothRelease:
+        release_date = make_release_date(**kwargs)
+        release = ThothRelease(dag_id=self.dag_id, release_date=release_date)
+        self.download_folder = make_workflow_folder(self.dag_id, release_date, SubFolder.downloaded.value)
+        self.transform_folder = make_workflow_folder(self.dag_id, release_date, SubFolder.transformed.value)
```

Comment:
I would modify this a little bit, so that the folder created by `get_data_path(), dag_id, f"{dag_id}_{release_string}"` can be accessed separately, e.g. as self.workflow_folder.

Perhaps then `make_workflow_folder` could just create the path at `workflow_folder` and you can make the download_folder and transform_folder after this with os.path.join.

This is so that you can use `self.workflow_folder` to cleanup the whole workflow folder.

Reply:
What would you think of the following as a solution?:
```suggestion
def make_workflow_folder(dag_id: str, release_date: pendulum.DateTime, *subdirs: str) -> str:
    release_string = release_date.format("YYYY_MM_DD")
    path = os.path.join(get_data_path(), dag_id, f"{dag_id}_{release_string}", *subdirs)
    os.makedirs(path, exist_ok=True)
    return path
```

Reply:
Yeah that looks good